### PR TITLE
#162948999 Users should favourite and unfavourite articles

### PR DESCRIPTION
**Pull request by Nicanor008**
#### What does this PR do?
This PR merges the favourite articles feature branch to develop

#### Description of Task to be completed?
- Allow authenticated users to favourite and unfavourite articles

#### How should this be manually tested?
- Run the server `python manage.py runserver`. 
- Switch this branch `git checkout ft-favorite-article-162948999`. 
- Run the following PUT endpoint in postman ` http://127.0.0.1:8000/api/v1/articles/<slug>/favorite/` and ` http://127.0.0.1:8000/api/v1/favorites/` to get all favourites for the logged in user

#### What are the relevant pivotal tracker stories?
[162948999 Users should be able to favorite articles](https://www.pivotaltracker.com/story/show/162948999)

#### Screenshots 
*Favourite an article*
screen shot 2019-01-29 at 10 05 44

*Unfavourite an article*
screen shot 2019-01-29 at 10 06 08

*Favourites articles for the logged in user*
screen shot 2019-01-29 at 15 21 29

### CODE REVIEW

**Review by Nyakaru — COMMENTED** [review]
@Nicanor008 On the section of what does this PR do, maybe you could write something more descriptive enough.

File: authors/apps/articles/tests/test_favorite_articles.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        response = self.client.put(
+            url,
+            format='json',
+            HTTP_AUTHORIZATION='Token ' + token
+        )
+        return response
+
+    def test_favorite_Article(self):
+        token = self.login_user()
+        url = self.create_list_article_url
```
Comment (eduhmik) [inline authors/apps/articles/tests/test_favorite_articles.py]: @Nicanor008 line 30 - 33 can be replaced by just calling slug = self.create_article() from our base test

**Comment by Nicanor008** [comment]
@Nyakaru I have done some edits.

File: authors/apps/articles/tests/test_favorite_articles.py (hunk 1 above)
Comment (Nicanor008) [reply authors/apps/articles/tests/test_favorite_articles.py]: Checking this out @eduhmik

Comment (Nicanor008) [reply authors/apps/articles/tests/test_favorite_articles.py]: Feedback implemented

**Review by kzyangiro — CHANGES_REQUESTED** [review]
@Nicanor008 Only push migration files that are relevant to your work, the rest never push. Kindly remove those not related.

**Comment by Nicanor008** [comment]
> @Nicanor008 Only push migration files that are relevant to your work, the rest never push. Kindly remove those not related.

@kzyGit after rebase from develop, I dropped all migrations because I was encountering server running errors. Also minus pushing them, builds failed on Travis.

File: authors/apps/articles/tests/test_favorite_articles.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            HTTP_AUTHORIZATION='Token ' + token
+        )
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_invalid_token_favorites_articles(self):
+        token = 'sdsfsdfsd'
+        response = self.client.get(
+            self.my_favorites_url(),
+            format='json',
+            HTTP_AUTHORIZATION='Token ' + token
```
Comment (charitymarani) [inline authors/apps/articles/tests/test_favorite_articles.py:56]: I feel  the block  containing `HTTP_AUTHORIZATION='Token ' + token` is redundant.I have actually created a function to authorize a user in the base_test.py.You could check it out in my PR for bookmarking. And also the status codes are redundant,200_Ok is appearing quite a number of times. Look into Dryness of your tests

Comment (Nicanor008) [reply authors/apps/articles/tests/test_favorite_articles.py:56]: Thanks for the feedback @charitymarani, implemented

**Review by kzyangiro — APPROVED** [review]

**Pull request merged**